testsuite/classlibrary/TestTempoClock.sc

```diff
@@ -0,0 +1,94 @@
+TestTempoClock : UnitTest {
+	var clock;
+
+	setUp {
+		clock = TempoClock.new(1000);
+	}
+
+	// called after each test
+	tearDown {
+		clock.stop;
+	}
+
+	test_isRunning {
+		var beforeStop = clock.isRunning, afterStop;
+		clock.stop;
+		afterStop = clock.isRunning;
+		this.assert(beforeStop and: { afterStop.not }, "TempoClock:isRunning should be true before 'stop', and false afterward");
+	}
+
+	test_afterStop_isRemovedFromAll {
+		clock.stop;
+		this.assert(TempoClock.all.includes(clock).not, "TempoClock.all should not contain a stopped instance");
+	}
+
+	test_threadWait_advancesCorrectTime {
+		var now = SystemClock.seconds, then, cond = Condition.new;
+		{
+			1.wait;
+			then = SystemClock.seconds;
+			cond.unhang;
+		}.fork(clock);
+		cond.hang;
+		// note: floating point math, == is not safe
+		this.assert((then - now absdif: 0.001) < 1e-10, "TempoClock 1000 beats/sec, 1 beat should be 0.001 seconds");
+	}
+
+	test_beats2secs_validConversion {
+		this.assert(
+			clock.beats2secs(clock.beats + 1) - SystemClock.seconds absdif: 0.001 < 1e-10,
+			"TempoClock 1000 beats/sec, 'beats2secs' for now + 1 beat should be 0.001 seconds later"
+		)
+	}
+
+	test_secs2beats_validConversion {
+		this.assert(
+			clock.secs2beats(SystemClock.seconds + 1) - clock.beats absdif: 1000.0 < 1e-10,
+			"TempoClock 1000 beats/sec, 'secs2beats' for now + 1 second should be 1000 beats later"
+		)
+	}
+
+	test_defaultMeter_unchanged {
+		this.assert(
+			clock.beatsPerBar == 4,
+			"A new TempoClock should have 4 beats per bar by default"
```

Comment:
use assertEquals, change method name to `test_beatsPerBar_defaultIs4` and remove message

Reply:
may want to turn this into a test for multiple properties of a default constructed tempoclock if there are other things you could check